programs/utilities/flower.sc

```diff
@@ -0,0 +1,136 @@
+//
+// Places the flower pattern that bonemeal creates when in flower forests.
+// large areas may cause quite a bit of lag to run.
+// Only places grass blocks + flowers in flower forest biomes.
+//
+// command usage:
+// /flower <position> print - prints what flower will generate at given position
+// /flower <position> create <radius> - places grass block with flowers on top in given radius (square area)
+// /flower <position> create <radius> only <flower> - same as create, but only places the chosen flower type
+//
+
+// flowers that generate in flower forest biome (in order)
+global_FLOWERS = ['dandelion', 'poppy', 'allium', 'azure_bluet', 'red_tulip', 'orange_tulip', 'white_tulip', 'pink_tulip', 'oxeye_daisy', 'cornflower', 'lily_of_the_valley'];
+
+__config() -> {
+  'stay_loaded' -> true,
+  'commands' -> {
+    '<pos> print' -> _(position) -> print(get_flower_at(position)),
+    '<pos> create <radius>' -> 'flower_area',
+    '<pos> create <radius> only <flower>' -> 'filtered_flower_area'
+  },
+  'arguments' -> {
+    'pos' -> {
+      'type' -> 'pos'
+    },
+    'radius' -> {
+      'type' -> 'int',
+      'min' -> 0,
+      'max' -> 1000,
+      'suggest' -> [10, 50, 100, 250]
+    },
+    'flower' -> {
+      'type' -> 'block',
+      'suggest' -> global_FLOWERS
+    }
+  }
+};
+
+// fills area of given radius with flowers
+flower_area(position, radius) -> (
+  count = 0;
+  without_updates(
+    scan(position, radius, 0, radius,
+      count += __flower_forest_fill(_);
+    );
+  );
+  print('Filled ' + count + ' flowers');
+);
+
+// fills area of given radius with flowers of given type
+filtered_flower_area(position, radius, flower_filter) -> (
+  // removes "minecraft:" in flower name if exists
+  fixed_flower_name = flower_filter - 'minecraft:';
+  flower_id = global_FLOWERS ~ fixed_flower_name;
+
+  // checks if given filter is a flower
+  if(flower_id == null,
+    print('not a valid flower: ' + flower_filter);
+    return();
+  );
+
+  count = 0;
+  without_updates(
+    scan(position, radius, 0, radius,
+      count += __filtered_flower_forest_fill(_, fixed_flower_name, flower_id);
+    );
+  );
+  print('Filled ' + count + ' flowers');
+);
+
+_get_flower_id_at(position) -> (
+  x = position : 0;
+  z = position : 2;
+
+  // gets simplex noise for given  coordinate with seed 2345 (from vanilla code)
+  val = simplex(x / 24, z / 24, null, 2345);
+  // makes sure output is between 0 and 0.9999
+  val_clamped = min(max(val, 0), 0.9999);
+  flower_id = floor(val_clamped * length(global_FLOWERS));
+
+  return(flower_id);
+);
+
+get_flower_at(position) -> (
+  flower_id = _get_flower_id_at(position);
+  return(global_FLOWERS : flower_id);
+);
+
+// check for flower forest biome
+_is_flower_forest(position) -> (
+  return(biome(position) == 'flower_forest');
```

Comment:
try to avoid unnecessary return calls, cos they tend to be slow in scarpet specifically. In this case, you can just do
```
_is_flower_forest(position) ->
  biome(position) == 'flower_forest';
```